src/Parser/Unlit.idr

```diff
@@ -36,16 +38,50 @@ styleTeX = MkLitStyle
               Nil
               [".tex", ".ltx"]
 
+
+||| Return the list of extensions used for literate files.
+export
+listOfExtensionsLiterate : List String
+listOfExtensionsLiterate
+  = concatMap file_extensions
+              [ styleBird
+              , styleOrg
+              , styleCMark
+              , styleTeX
+              ]
+
+||| Are we dealing with a valid literate file name, if so return the base name and used extension.
+export
+hasLitFileExt : (fname : String)
+                     -> Maybe (String, String)
+hasLitFileExt fname
+      = find listOfExtensionsLiterate
+
+  where
+    rp : String -> List Char
+    rp = (reverse . unpack)
+
+    pr : List Char -> String
+    pr = (pack . reverse)
+
+    find : List String -> Maybe (String, String)
+    find Nil
+      = Nothing
+
+    find (ext::xs)
+      = if (isSuffixOf fname ext)
+          then let bname_r = (deleteFirstsBy (==) (rp fname) (rp ext))
+               in Just (pr bname_r, ext)
+          else find xs
```

Comment:
I thought you would say that! The reason for the convoluted operation is to provide future support for literate files of the form: `<literate mode>.idr`. This is supported by agda, and perhaps something we should support here.

The path based operation, whilst perfectly valid, removes the very last extension. This would not work with the multi-dot extensions like `Foo.org.idr`.

Reply:
I've added new library functions and pushed a refactoring using them. Please review!

Reply:
The semantics of the changes (better filename splitting) looks spot on. Thanks for cleaning that small wart up. I suppose what is next is to add new file extensions for recognition to test it out.